common/persistence/persistence-tests/executionManagerTest.go

```diff
@@ -2265,6 +2265,119 @@ func (s *ExecutionManagerSuite) TestWorkflowReplicationState() {
+	s.NoError(err0)
+	s.NotNil(task0, "Expected non empty task identifier.")
+
+	stats0, state0, err1 := s.GetWorkflowExecutionInfoWithStats(domainID, workflowExecution)
+	s.NoError(err1)
+	info0 := state0.ExecutionInfo
+	s.NotNil(info0, "Valid Workflow info expected.")
+	s.Equal(0, stats0.BufferedEventsCount)
+	s.Equal(0, stats0.BufferedEventsSize)
+
+	eventsBatch1 := []*gen.HistoryEvent{
+		&gen.HistoryEvent{
+			EventId:   common.Int64Ptr(5),
+			EventType: gen.EventTypeDecisionTaskCompleted.Ptr(),
+			Version:   common.Int64Ptr(11),
+			DecisionTaskCompletedEventAttributes: &gen.DecisionTaskCompletedEventAttributes{
+				ScheduledEventId: common.Int64Ptr(2),
+				StartedEventId:   common.Int64Ptr(3),
+				Identity:         common.StringPtr("test_worker"),
+			},
+		},
+		&gen.HistoryEvent{
+			EventId:   common.Int64Ptr(6),
+			EventType: gen.EventTypeTimerStarted.Ptr(),
+			Version:   common.Int64Ptr(11),
+			TimerStartedEventAttributes: &gen.TimerStartedEventAttributes{
+				TimerId:                      common.StringPtr("ID1"),
+				StartToFireTimeoutSeconds:    common.Int64Ptr(101),
+				DecisionTaskCompletedEventId: common.Int64Ptr(5),
+			},
+		},
+	}
+	bufferedTask1 := &p.BufferedReplicationTask{
+		FirstEventID: int64(5),
+		NextEventID:  int64(7),
+		Version:      int64(11),
+		History:      eventsBatch1,
+	}
+
+	eventsBatch2 := []*gen.HistoryEvent{
+		&gen.HistoryEvent{
+			EventId:   common.Int64Ptr(21),
+			EventType: gen.EventTypeTimerFired.Ptr(),
+			Version:   common.Int64Ptr(12),
+			TimerFiredEventAttributes: &gen.TimerFiredEventAttributes{
+				TimerId:        common.StringPtr("2"),
+				StartedEventId: common.Int64Ptr(3),
+			},
+		},
+	}
+	bufferedTask2 := &p.BufferedReplicationTask{
+		FirstEventID: int64(21),
+		NextEventID:  int64(22),
+		Version:      int64(12),
+		History:      eventsBatch2,
+	}
+	updatedInfo := copyWorkflowExecutionInfo(info0)
+	updatedState := &p.WorkflowMutableState{
+		ExecutionInfo: updatedInfo,
+	}
+
+	err2 := s.UpdateAllMutableState(updatedState, int64(3))
+	s.NoError(err2)
+
+	partialState, err2 := s.GetWorkflowExecutionInfo(domainID, workflowExecution)
+	s.NoError(err2)
+	s.NotNil(partialState, "expected valid state.")
+	partialInfo := partialState.ExecutionInfo
+	s.NotNil(partialInfo, "Valid Workflow info expected.")
+
+	bufferUpdateInfo := copyWorkflowExecutionInfo(partialInfo)
+	err2 = s.UpdateWorklowStateAndReplication(bufferUpdateInfo, nil, bufferedTask1, nil, bufferUpdateInfo.NextEventID, nil)
+	s.NoError(err2)
+	err2 = s.UpdateWorklowStateAndReplication(bufferUpdateInfo, nil, bufferedTask2, nil, bufferUpdateInfo.NextEventID, nil)
+	s.NoError(err2)
+	err2 = s.UpdateWorkflowExecutionForBufferEvents(bufferUpdateInfo, nil, bufferUpdateInfo.NextEventID, eventsBatch1, false)
+	s.NoError(err2)
+	stats0, _, err2 = s.GetWorkflowExecutionInfoWithStats(domainID, workflowExecution)
+	s.NoError(err2)
+	s.Equal(1, stats0.BufferedEventsCount)
```

Comment:
Can you also add assertion on size.

Reply:
Added more assertion to compare the history events after update.
But the historySize is not stable. Looks like thriftRW doesn't guarantee that:

```
longer@~/gocode/src/github.com/uber/cadence:(buffered_event_unit_test)$ grep 'stats1.BufferedEventsSize' mytest.log.*
mytest.log.0:stats1.BufferedEventsSize 454
mytest.log.0:stats1.BufferedEventsSize 213
mytest.log.1:stats1.BufferedEventsSize 213
mytest.log.1:stats1.BufferedEventsSize 400
mytest.log.2:stats1.BufferedEventsSize 213
mytest.log.2:stats1.BufferedEventsSize 267
longer@~/gocode/src/github.com/uber/cadence:(buffered_event_unit_test)$
longer@~/gocode/src/github.com/uber/cadence:(buffered_event_unit_test)$
longer@~/gocode/src/github.com/uber/cadence:(buffered_event_unit_test)$ grep 'stats0.BufferedEventsSize' mytest.log.*
mytest.log.0:stats0.BufferedEventsSize 337
mytest.log.0:stats0.BufferedEventsSize 150
mytest.log.1:stats0.BufferedEventsSize 150
mytest.log.1:stats0.BufferedEventsSize 337
mytest.log.2:stats0.BufferedEventsSize 150
mytest.log.2:stats0.BufferedEventsSize 150
```

Reply:
Let's just put a assertion to check size greater than 0.